modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/ImageMosaicDatastoreWalker.java

```diff
@@ -233,4 +257,151 @@ public void run() {
             }
         }
     }
+
+    protected void handleCogGranule(final String granuleBeingProcessed) throws IOException {
```

Comment:
So in the end the handling of COG refs and normal files was too different to homogeneize eh?

Reply:
yep. there was several file-centric checks and assumptions here and there.